### PR TITLE
enh(UI): Update React Router

### PR DESCRIPTION
**Pull request by Thebarda**
## Description

This migrates React Router to version 6.
This also refactor some legacy code.
NOTE: This is the first PR to refactor the poller wizard. The next PR will remove the redirection between steps. 

https://user-images.githubusercontent.com/12515407/145115601-1c177ca9-d6f4-4991-a79a-46147f5695e7.mov

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [x] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 20.10.x
- [ ] 21.04.x
- [ ] 21.10.x
- [x] 22.04.x (master)

<h2> How this pull request can be tested ? </h2>

For each redirection, only the page content (the below the top counter and beside the navigation menu) should change.
- Try redirection using the menu
- Try the redirection using the top counter (don't forget to test the deprecated pages option)
- Try the redirection from the Resource status graph to Performance graph page
- Try the automatic redirection when the user session ends

## Checklist

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: www/front_src/src/App.tsx [hunk 1]
```diff
@@ -64,154 +63,102 @@ const styles = createStyles({
...
+    keepAlive();
+
+    keepAliveIntervalRef.current = setInterval(keepAlive, 15000);
+  }, []);
+
+  const min = hasMinArgument();
+
+  return (
+    <React.Suspense fallback={<PageLoader />}>
+      <ThemeProvider>
```
Comment (bdauria) [inline www/front_src/src/App.tsx:117]: Shouldn't we move the ThemeProvider to our Provider Component?

File: www/front_src/src/Header/RessourceStatusCounter/Host/index.tsx [hunk 2]
```diff
@@ -67,43 +72,86 @@ interface HostData {
...
+    (e): void => {
+      e.preventDefault();
+      toggle?.();
+      if (use_deprecated_pages) {
+        navigate(link);
+
+        return;
+      }
+      applyFilter({ criterias, id: '', name: 'New Filter' });
+      navigate(link);
```
Comment (bdauria) [inline www/front_src/src/Header/RessourceStatusCounter/Host/index.tsx:152]:
```suggestion
      if (!use_deprecated_pages) {
        applyFilter({ criterias, id: '', name: 'New Filter' });
      }

      navigate(link);
```

File: www/front_src/src/Header/RessourceStatusCounter/Service/index.tsx [hunk 3]
```diff
@@ -58,48 +63,96 @@ const statusSchema = yup.object().shape({
...
+    (e): void => {
+      e.preventDefault();
+      toggle?.();
+      if (use_deprecated_pages) {
+        navigate(link);
+
+        return;
+      }
+      applyFilter({ criterias, id: '', name: 'New Filter' });
+      navigate(link);
```
Comment (bdauria) [inline www/front_src/src/Header/RessourceStatusCounter/Service/index.tsx:153]:
```suggestion
      if (use_deprecated_pages) {
        applyFilter({ criterias, id: '', name: 'New Filter' });
      }

      navigate(link);
```

File: www/front_src/src/Resources/Filter/useFilter.ts [hunk 4]
```diff
@@ -99,7 +96,7 @@ const useFilter = (): void => {
       },
     ]);
 
-    applyFilter(defaultFilter);
+    // applyFilter(defaultFilter());
```
Comment (bdauria) [inline www/front_src/src/Resources/Filter/useFilter.ts:99]: Remove comment

File: www/front_src/src/App.tsx (hunk 1 above)
Comment (Thebarda) [reply www/front_src/src/App.tsx:117]: ThemeProvider is already defined in the Provider component. I'll remove it

**Review by jdelpierre — APPROVED** [review]

**Review by Jordan91 — APPROVED** [review]

**Review by a-launois — APPROVED** [review]

**Review by bdauria — APPROVED** [review]

**Pull request merged**